cmd/relay/main.go

```diff
@@ -109,9 +110,16 @@ func Relay(c *cli.Context) error {
 		return err
 	}
 	fmt.Printf("Listening at %s\n", listener.Addr())
+	go checkHealth(listener.Addr())
 	return http.Serve(listener, handler)
 }
 
+func checkHealth(addr net.Addr) {
+	time.Sleep(10 * time.Millisecond)
+	client := http.Client{}
+	client.Get(fmt.Sprintf("http://%s/public/0", addr.String()))
+}
+
```

Comment:
humm this is ugly... checkHealth could theoretically run before http.Serve()? Can't you interact with the handler directly or do this in a cleaner way than calling itself?